agent/command/macsign.go

```diff
@@ -0,0 +1,232 @@
+	LocalZipFile string `mapstructure:"local_zip_file" plugin:"expand"`
+
+	// OutputZipFile is the local filepath to the zip file the service outputs
+	// It will contain the list of artifacts that are signed by the server.
+	OutputZipFile string `mapstructure:"output_zip_file" plugin:"expand"`
+
+	// ArtifactType is a type of artifact(s) that need to be signed.
+	// Currently supported list: app, binary.
+	ArtifactType string `mapstructure:"artifact_type" plugin:"expand"`
+
+	// EntitlementsFilePath is the local filepath to the entitlements file that the users
+	// wishes to execute the signing process with. This is optional.
+	EntitlementsFilePath string `mapstructure:"entitlements_file" plugin:"expand"`
+
+	// Verify determines if the signature(or notarization) should be verified.
+	// Verification is only supported on MacOS. It is optional, default value if false.
+	Verify bool `mapstructure:"verify"`
+
+	// Notarize determines if the file should also be notarized after signing.
+	Notarize bool `mapstructure:"notarize"`
+
+	// BundleId is the bundle id of the artifact used during notarization.
+	// This is mandatory if notarization is requested.
+	BundleId string `mapstructure:"bundle_id"  plugin:"expand"`
+
+	// BuildVariants stores a list of MCI build variants to run the command for.
+	// If the list is empty, it runs for all build variants.
+	BuildVariants []string `mapstructure:"build_variants"`
+
+	// WorkingDir sets the current working directory.
+	WorkingDir string `mapstructure:"working_directory" plugin:"expand"`
+
+	base
+}
+
+func macSignFactory() Command         { return &macSign{} }
+func (macSign *macSign) Name() string { return "mac.sign" }
+
+// macSign-specific implementation of ParseParams.
+func (macSign *macSign) ParseParams(params map[string]interface{}) error {
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Result: macSign,
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err := decoder.Decode(params); err != nil {
+		return errors.Wrapf(err, "error decoding %s params", macSign.Name())
+	}
+
+	return macSign.validate()
+}
+
+func (macSign *macSign) validate() error {
+	catcher := grip.NewSimpleCatcher()
+
+	// make sure the command params are valid
+	if macSign.KeyId == "" {
+		catcher.New("key_id cannot be blank")
+	}
+	if macSign.Secret == "" {
+		catcher.New("secret cannot be blank")
+	}
+	if macSign.LocalZipFile == "" {
+		catcher.New("local_zip_file cannot be blank")
+	}
+	if macSign.OutputZipFile == "" {
+		catcher.New("output_zip_file cannot be blank")
+	}
+	if macSign.ServiceUrl == "" {
+		catcher.New("service_url cannot be blank")
+	}
+	if runtime.GOOS != "darwin" {
+		// do not fail, just set verifying to false.
+		macSign.Verify = false
+	}
+	if !(macSign.ArtifactType == "" || macSign.ArtifactType == "binary" || macSign.ArtifactType == "app") {
+		catcher.New(`artifact_type needs to be either blank, "binary" or "app"`)
```

Comment:
consistency nit: we typically still use "" as outer quotes, and ' ' as inner quotes